modules/gdscript/gdscript_parser.cpp

```diff
@@ -6767,10 +6767,10 @@ GDScriptParser::DataType GDScriptParser::_reduce_function_call_type(const Operat
 				valid = _get_function_signature(base_type, callee_name, return_type, arg_types,
 						default_args_count, is_static, is_vararg);
 
-				if (valid) {
-					return_type = original_type;
-					return_type.is_meta_type = false;
-				}
+				return_type = original_type;
+				return_type.is_meta_type = false;
+
+				valid = true; // There's always an initializer, we can asume this is true
 			}
 
 			if (!valid) {
```

Comment:
As per the above, this condition could now simply be an `else` branch for the previous `if (is_initializer && original_type.is_meta_type))`. But it's likely better to keep it as is as `valid` is clearer and keeps being used down the road.